MediaBrowser.Providers/Plugins/TheTvdb/TvdbSeriesProvider.cs

```diff
@@ -75,27 +58,98 @@ public async Task<MetadataResult<Series>> GetMetadata(SeriesInfo itemId, Cancell
             }
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (IsValidSeries(itemId.ProviderIds))
+            if (IsValidSeries(itemId))
             {
                 result.Item = new Series();
                 result.HasMetadata = true;
 
-                await FetchSeriesData(result, itemId.MetadataLanguage, itemId.ProviderIds, cancellationToken)
+                await FetchSeriesMetadata(result, itemId.MetadataLanguage, itemId.ProviderIds, cancellationToken)
                     .ConfigureAwait(false);
             }
 
             return result;
         }
 
-        private async Task FetchSeriesData(MetadataResult<Series> result, string metadataLanguage, Dictionary<string, string> seriesProviderIds, CancellationToken cancellationToken)
+        private async Task<IEnumerable<RemoteSearchResult>> FetchSeriesSearchResult(SeriesInfo seriesInfo, CancellationToken cancellationToken)
+        {
+            var tvdbId = seriesInfo.GetProviderId(MetadataProvider.Tvdb);
+            if (string.IsNullOrEmpty(tvdbId))
+            {
+                var imdbId = seriesInfo.GetProviderId(MetadataProvider.Imdb);
+                if (!string.IsNullOrEmpty(imdbId))
+                {
+                    tvdbId = await GetSeriesByRemoteId(
+                        imdbId,
+                        MetadataProvider.Imdb.ToString(),
+                        seriesInfo.MetadataLanguage,
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            if (string.IsNullOrEmpty(tvdbId))
+            {
+                var zap2ItId = seriesInfo.GetProviderId(MetadataProvider.Zap2It);
+                if (!string.IsNullOrEmpty(zap2ItId))
+                {
+                    tvdbId = await GetSeriesByRemoteId(
+                        zap2ItId,
+                        MetadataProvider.Zap2It.ToString(),
+                        seriesInfo.MetadataLanguage,
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            try
+            {
+                var seriesResult =
+                    await _tvdbClientManager
+                        .GetSeriesByIdAsync(Convert.ToInt32(tvdbId), seriesInfo.MetadataLanguage, cancellationToken)
+                        .ConfigureAwait(false);
+                return new []{ MapSeriesToRemoteSearchResult(seriesResult.Data) };
+            }
+            catch (TvDbServerException e)
+            {
+                _logger.LogError(e, "Failed to retrieve series with id {TvdbId}", tvdbId);
+                return Array.Empty<RemoteSearchResult>();
+            }
+        }
+
+        private RemoteSearchResult MapSeriesToRemoteSearchResult(TvDbSharper.Dto.Series series)
+        {
+            var remoteResult = new RemoteSearchResult
+            {
+                Name = series.SeriesName,
+                Overview = (series.Overview ?? string.Empty).Trim(),
```

Comment:
Wouldn't this be better as it wouldnt result in the possibility that string.Empty got trimmed?

```suggestion
                Overview = series?.Overview.Trim() ?? string.Empty,
```